src/parser.jl

```diff
@@ -1169,19 +1169,18 @@ function parse_unary_call(ps::ParseState)
         mark_before_paren = position(ps)
         bump(ps, TRIVIA_FLAG) # (
         initial_semi = peek(ps) == K";"
-        is_call = Ref(false)
-        is_block = Ref(false)
-        parse_brackets(ps, K")") do had_commas, had_splat, num_semis, num_subexprs
-            is_call[] = had_commas || had_splat || initial_semi
-            is_block[] = !is_call[] && num_semis > 0
-            return (needs_parameters=is_call[],
-                    eq_is_kw_before_semi=is_call[],
-                    eq_is_kw_after_semi=is_call[])
+        opts = parse_brackets(ps, K")") do had_commas, had_splat, num_semis, num_subexprs
+            is_call = had_commas || had_splat || initial_semi
+            return (needs_parameters=is_call,
+                    eq_is_kw_before_semi=is_call,
+                    eq_is_kw_after_semi=is_call,
+                    is_call=is_call,
+                    is_block=!is_call && num_semis > 0)
         end
```

Comment:
In my recent allocations reduction PR (https://github.com/JuliaConcurrent/MultiThreadedCaches.jl/pull/10/), I also had to fight a closure that allocates, in order to capture a variable, and forces the variable to be Boxed.

I ended up being able to change the `get!(dict, key) do` pattern into `get(dict, key, :cache_miss)` and then use an if-else instead of a closure in the miss case.
I think it means it'll have to do two different lookups in the dictionary though, which is too bad. :(

Your solution here to return is_call instead of mutating it as a side-effect looks great 👍

Reply:
yes in the end I couldn't figure out how to do this with a side effect (even though the `do` block syntax makes that very convenient and natural when looking at the code).  Doing it with a return value was fine in this case because I can modify `parse_brackets` to actually return that return value.